diff --git a/node_modules/@react-native-firebase/messaging/ios/RNFBMessaging/RNFBMessaging+UNUserNotificationCenter.m b/node_modules/@react-native-firebase/messaging/ios/RNFBMessaging/RNFBMessaging+UNUserNotificationCenter.m
index 541114d..c578670 100644
--- a/node_modules/@react-native-firebase/messaging/ios/RNFBMessaging/RNFBMessaging+UNUserNotificationCenter.m
+++ b/node_modules/@react-native-firebase/messaging/ios/RNFBMessaging/RNFBMessaging+UNUserNotificationCenter.m
@@ -74,7 +74,7 @@ - (void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNot
     }
 
     // TODO in a later version allow customising completion options in JS code
-    completionHandler(UNNotificationPresentationOptionNone);
+//    completionHandler(UNNotificationPresentationOptionNone);
   }
 
   if (_originalDelegate != nil && originalDelegateRespondsTo.willPresentNotification) {
