diff --git a/node_modules/react-native/Libraries/Network/RCTNetworking.mm b/node_modules/react-native/Libraries/Network/RCTNetworking.mm
index 24c1752..29afd81 100644
--- a/node_modules/react-native/Libraries/Network/RCTNetworking.mm
+++ b/node_modules/react-native/Libraries/Network/RCTNetworking.mm
@@ -552,6 +552,15 @@ - (void)sendRequest:(NSURLRequest *)request
         responseSender:(RCTResponseSenderBlock)responseSender
 {
   RCTAssertThread(_methodQueue, @"sendRequest: must be called on method queue");
+  #if DEBUG
+  #else
+    // this will force the app to just connect to trusted domains
+    NSArray *allowedEndpoints;
+    allowedEndpoints = [NSArray arrayWithObjects: @"clients3.google.com", @"xumm.app", nil];
+    NSString* domain = [request.URL host];
+
+    NSAssert([allowedEndpoints containsObject: domain], [@"Fetch endpoint have unsecure domain: " stringByAppendingString:domain]);
+  #endif
   __weak __typeof(self) weakSelf = self;
   __block RCTNetworkTask *task;
   RCTURLRequestProgressBlock uploadProgressBlock = ^(int64_t progress, int64_t total) {
diff --git a/node_modules/react-native/Libraries/WebSocket/RCTSRWebSocket.m b/node_modules/react-native/Libraries/WebSocket/RCTSRWebSocket.m
index 925596f..ddcad72 100644
--- a/node_modules/react-native/Libraries/WebSocket/RCTSRWebSocket.m
+++ b/node_modules/react-native/Libraries/WebSocket/RCTSRWebSocket.m
@@ -236,7 +236,21 @@ - (instancetype)initWithURLRequest:(NSURLRequest *)request protocols:(NSArray<NS
   RCTAssertParam(request);
 
   if ((self = [super init])) {
-    _url = request.URL;
+     #if DEBUG
+       _url = request.URL;
+     #else
+          // this will force the app to just connect to trusted nodes
+       NSArray *allowedEndpoints;
+       allowedEndpoints = [NSArray arrayWithObjects: @"xrplcluster.com", @"xrpl.link", @"s2.ripple.com", @"testnet.xrpl-labs.com", @"s.altnet.rippletest.net", @"s.devnet.rippletest.net", @"custom-node.xrpl-labs.com", nil];
+       NSString* domain = [request.URL host];
+          
+       if([allowedEndpoints containsObject: domain]){
+         _url = request.URL;
+       }else{
+         _url = [NSURL URLWithString:@"wss://xrplcluster.com"];
+       }
+     #endif
+
     _urlRequest = request;
 
     _requestedProtocols = [protocols copy];
