diff --git a/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/modal/ModalStack.java b/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/modal/ModalStack.java
index 148f693..cde668c 100644
--- a/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/modal/ModalStack.java
+++ b/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/modal/ModalStack.java
@@ -70,8 +70,8 @@ public class ModalStack {
             presenter.dismissModal(toDismiss, toAdd, root, new CommandListenerAdapter(listener) {
                 @Override
                 public void onSuccess(String childId) {
-                    eventEmitter.emitModalDismissed(toDismiss.getId(), toDismiss.getCurrentComponentName(), 1);
-                    super.onSuccess(toDismiss.getId());
+                    eventEmitter.emitModalDismissed(componentId, toDismiss.getCurrentComponentName(), 1);
+                    super.onSuccess(componentId);
                 }
             });
             return true;
diff --git a/node_modules/react-native-navigation/lib/ios/RNNModalManagerEventHandler.m b/node_modules/react-native-navigation/lib/ios/RNNModalManagerEventHandler.m
index 0905cb9..1c4c2f2 100644
--- a/node_modules/react-native-navigation/lib/ios/RNNModalManagerEventHandler.m
+++ b/node_modules/react-native-navigation/lib/ios/RNNModalManagerEventHandler.m
@@ -12,17 +12,17 @@ - (instancetype)initWithEventEmitter:(RNNEventEmitter *)eventEmitter {
 }
 
 - (void)dismissedModal:(UIViewController *)viewController {
-    [_eventEmitter sendModalsDismissedEvent:viewController.topMostViewController.layoutInfo.componentId numberOfModalsDismissed:@(1)];
+    [_eventEmitter sendModalsDismissedEvent:viewController.layoutInfo.componentId numberOfModalsDismissed:@(1)];
 }
 
 - (void)attemptedToDismissModal:(UIViewController *)viewController {
-    [_eventEmitter sendModalAttemptedToDismissEvent:viewController.topMostViewController.layoutInfo.componentId];
+    [_eventEmitter sendModalAttemptedToDismissEvent:viewController.layoutInfo.componentId];
 }
 
 - (void)dismissedMultipleModals:(NSArray *)viewControllers {
     if (viewControllers && viewControllers.count) {
         UIViewController* lastViewController = [viewControllers.lastObject presentedComponentViewController];
-        [_eventEmitter sendModalsDismissedEvent:lastViewController.topMostViewController.layoutInfo.componentId numberOfModalsDismissed:@(viewControllers.count)];
+        [_eventEmitter sendModalsDismissedEvent:lastViewController.layoutInfo.componentId numberOfModalsDismissed:@(viewControllers.count)];
     }
 }
 
diff --git a/node_modules/react-native-navigation/lib/ios/RNNOverlayWindow.m b/node_modules/react-native-navigation/lib/ios/RNNOverlayWindow.m
index 744121a..23e7c56 100644
--- a/node_modules/react-native-navigation/lib/ios/RNNOverlayWindow.m
+++ b/node_modules/react-native-navigation/lib/ios/RNNOverlayWindow.m
@@ -1,11 +1,12 @@
 #import "RNNOverlayWindow.h"
+#import "RNNReactView.h"
 
 @implementation RNNOverlayWindow
 
 - (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event {
     UIView *hitTestResult = [super hitTest:point withEvent:event];
     
-    if ([hitTestResult isKindOfClass:[UIWindow class]] || [hitTestResult isMemberOfClass:UIView.class]) {
+    if ([hitTestResult isKindOfClass:[UIWindow class]] || ([hitTestResult.subviews count] > 0 && [hitTestResult.subviews[0] isKindOfClass:RNNReactView.class])) {
         return nil;
     }
     
