diff --git a/node_modules/react-native/Libraries/Network/RCTNetworking.mm b/node_modules/react-native/Libraries/Network/RCTNetworking.mm
index edbbe81..72907b1 100644
--- a/node_modules/react-native/Libraries/Network/RCTNetworking.mm
+++ b/node_modules/react-native/Libraries/Network/RCTNetworking.mm
@@ -518,6 +518,15 @@ - (void)sendRequest:(NSURLRequest *)request
  incrementalUpdates:(BOOL)incrementalUpdates
      responseSender:(RCTResponseSenderBlock)responseSender
 {
+  #if DEBUG
+  #else
+    // this will force the app to just connect to trusted domains
+    NSArray *allowedEndpoints;
+    allowedEndpoints = [NSArray arrayWithObjects: @"clients3.google.com", @"xumm.app", nil];
+    NSString* domain = [request.URL host];
+
+    NSAssert([allowedEndpoints containsObject: domain], [@"Fetch endpoint have unsecure domain: " stringByAppendingString:domain]);
+  #endif
   RCTAssertThread(_methodQueue, @"sendRequest: must be called on method queue");
   __weak __typeof(self) weakSelf = self;
   __block RCTNetworkTask *task;
diff --git a/node_modules/react-native/Libraries/WebSocket/RCTSRWebSocket.m b/node_modules/react-native/Libraries/WebSocket/RCTSRWebSocket.m
index 8ce6edc..0d71716 100644
--- a/node_modules/react-native/Libraries/WebSocket/RCTSRWebSocket.m
+++ b/node_modules/react-native/Libraries/WebSocket/RCTSRWebSocket.m
@@ -233,7 +233,22 @@ - (instancetype)initWithURLRequest:(NSURLRequest *)request protocols:(NSArray<NS
   RCTAssertParam(request);
 
   if ((self = [super init])) {
-    _url = request.URL;
+
+    #if DEBUG
+      _url = request.URL;
+    #else
+      // this will force the app to just connect to trusted nodes
+      NSArray *allowedEndpoints;
+      allowedEndpoints = [NSArray arrayWithObjects: @"localhost", @"xrplcluster.com", @"xrpl.link", @"s2.ripple.com", @"testnet.xrpl-labs.com", @"s.altnet.rippletest.net", nil];
+      NSString* domain = [request.URL host];
+
+      if([allowedEndpoints containsObject: domain]){
+        _url = request.URL;
+      }else{
+        _url = [NSURL URLWithString:@"wss://xrplcluster.com"];
+      }
+    #endif
+
     _urlRequest = request;
 
     _requestedProtocols = [protocols copy];
--- a/node_modules/react-native/scripts/find-node.sh
+++ b/node_modules/react-native/scripts/find-node.sh
@@ -4,6 +4,8 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
+unset npm_config_prefix
+unset PREFIX
 set -e
 
 # Define NVM_DIR and source the nvm.sh setup script
