diff --git a/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/component/ComponentViewController.java b/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/component/ComponentViewController.java
index 3117dcc..863f7cd 100644
--- a/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/component/ComponentViewController.java
+++ b/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/component/ComponentViewController.java
@@ -139,7 +139,7 @@ public class ComponentViewController extends ChildController<ComponentLayout> {
     protected WindowInsetsCompat applyWindowInsets(ViewController<?> view, WindowInsetsCompat insets) {
         ViewCompat.onApplyWindowInsets(view.getView(), insets.replaceSystemWindowInsets(
                 insets.getSystemWindowInsetLeft(),
-                insets.getSystemWindowInsetTop(),
+                Math.max(insets.getSystemWindowInsetTop() - getTopInset(), 0),
                 insets.getSystemWindowInsetRight(),
                 Math.max(insets.getSystemWindowInsetBottom() - getBottomInset(), 0)
         ));
diff --git a/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/overlay/OverlayManager.kt b/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/overlay/OverlayManager.kt
index 7ed36cb..9c71fdc 100644
--- a/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/overlay/OverlayManager.kt
+++ b/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/overlay/OverlayManager.kt
@@ -43,7 +43,10 @@ class OverlayManager {
     }
 
     fun destroy(overlaysContainer: ViewGroup) {
-        overlayRegistry.values.forEach { overlay -> destroyOverlay(overlaysContainer, overlay) }
+        val overlays: MutableList<ViewController<*>> = overlayRegistry.values.toMutableList()
+        for(overlay in overlays) {
+            destroyOverlay(overlaysContainer, overlay)
+        }
     }
 
     fun size() = overlayRegistry.size
